lib/format_parser.rb

```diff
@@ -8,6 +8,7 @@ module FormatParser
   require_relative 'remote_io'
   require_relative 'io_constraint'
   require_relative 'care'
+  require 'set'
```

Comment:
Since this is a standard library require (ships with Ruby) I would put it before the relative requires